### ISSUE TITLE
On logging

### ISSUE THREAD

**Issue by 097115** [mcpcpc/kirc#95]
When using `-o` option, `kirc` seems to handle line breaks in a strange way: it inserts `^M` followed by an empty line, like this (the log file opened in Vim):

![line-breaks](https://i.imgur.com/t6msZL3.png)

Is this an intended behaviour? :)

v0.2.2 running on Linux.

**Comment by mcpcpc** [comment]
Each line from the raw IRC stream is terminated by a carriage return and a line feed character (frequently referred to as `CR-LF`).  **kirc** prints the entire line from the stream without modifying it, including `CR-LF`.  In **vim**, the `^M` character is the carriage return character by default. So printing this character is intended behavior. 

In most practical applications for output logging, the raw steam data will be parsed/modified; therefore, printing this character will cause little interference to the end-user.  I am open to removing these characters though if you have particular application in mind where this would cause problems.  If you need a quick short-term solution though, you can always use the following **vim** command to remove it from your log file:

```
# you can insert the ^M using Ctrl+V and Ctrl+M keys
:s^M$//
```

Hope this information helps.

**Comment by kdkasad** [comment]
There also appears to to be a duplicate line feed (causing the blank line between printed lines), which can be fixed by removing the one also printed by the _fprintf()_ function:
> 
> https://github.com/mcpcpc/kirc/blob/9370902c898d6a51b505a947150ef3fff72af3ca/kirc.c#L397

I think this change should be implemented, since IRC commands already end with a newline. Plus, if someone is trying to use the log to debug an IRC command that does not end with a newline, they may think that it does since the print statement appends one, which could lead to problems.

**Comment by mcpcpc** [comment]
The extra new line feed has been removed per https://github.com/mcpcpc/kirc/pull/96.

 I think it makes sense to leave the `CR-LF` for the same argument you made above regarding debugging (since it is “expected” per the IRC protocol). I will merge the PR sometime tomorrow (after i have had a chance to consider a few additional changes) and 0.2.3 tarballs will be made available.

**Comment by 097115** [comment]
@mcpcpc Thanks for the info!

And while we are here, one more thought: may be the log entries should also contain timestamps, what do you think? When browsing a log from time to time to check for the missed conversations, timestamps could help to understand where exactly are we right now... (I'm sort of trying to use the log file as you've [suggested](https://github.com/mcpcpc/kirc/issues/93#issuecomment-731761456), grepping for #channel PRIVMSGs).

**Comment by mcpcpc** [comment]
> @mcpcpc Thanks for the info!
> 
> And while we are here, one more thought: may be the log entries should also contain timestamps, what do you think? When browsing a log from time to time to check for the missed conversations, timestamps could help to understand where exactly are we right now... (I'm sort of trying to use the log file as you've [suggested](https://github.com/mcpcpc/kirc/issues/93#issuecomment-731761456), grepping for #channel PRIVMSGs).

I was wondering when someone was going to request this.  lol.  Added per https://github.com/mcpcpc/kirc/commit/20a7e3bccd5d4332749f2cf672f16c2ab995cdc9 and will be released with 0.2.3 later today.

Format will be as such:

```
Fri Dec 18 13:34:55 2020::moon.freenode.net 372 kirc :- Thank you for using freenode!
```

**Comment by 097115** [comment]
:)

Thanks!
